modules/security/keyvault/main.bicep

```diff
@@ -0,0 +1,100 @@
+@description('Deployment Location')
+param location string
+
+@description('Prefix of Cosmos DB Resource Name')
+param prefix string = 'kv'
+
+@description('Name of the Key Vault')
+param name string = '${prefix}-${uniqueString(resourceGroup().id)}'
+
+@description('The tenant ID where the Key Vault is deployed')
+param tenantId string = subscription().tenantId
+
+@description('For an existing Managed Identity, the Subscription Id it is located in')
+param subscriptionId string = subscription().subscriptionId
+
+@description('For an existing Managed Identity, the Resource Group it is located in')
+param resourceGroupName string = resourceGroup().name
+
+@description('Subnet ID for the Key Vault')
+param subnetID string = ''
+
+@description('List of RBAC policies to assign to the Key Vault')
+param rbacPolicies array = []
```

Comment:
This may be a bit confusing to some considering RBAC assignment is not policy drive in KVs. It is just assigning an RBAC role. This array is being passed in to the nested module for role assignment and requires the Principal IDs that are to be assigned the given roles.

However, if I pass in multiple Principal IDs that means they are each going to be granted the default roleAssignments.

I would think it may be easier to create this as an array of objects that defined the Principal ID and the Role together. Example:

```
var rbacAssignment = [
    {
        principalId: '0000-000-000-00000-0000'
        role: 'Key Vault Secrest Officer'
    }
    {
        principalId: '00001-000-000-000-000000'
        role: 'Key Vault Secrets User'
    }
]
```

This lets them have more granular control over the RBAC assignments.

Reply:
The array you proposed makes sense. Let me think about the best way to bring this in. Custom types would make this better, but I think I can do a good job without.

Reply:
So, if you have similar to that, here is how you could use the module.

I think I am going to keep the module how it is, because I am having trouble cleaning implementing the conditional loops with the more complex objects.

I will include this as an example

```bicep
module keyVault 'br/public:security/keyvault:0.0.1' = {
  name: 'myKeyVault'
  params: {
    location: location
    name: name
  }
}

var rbacAssignments = [
    {
        principalIds: [ '0000-000-000-00000-0000' ]
        roles: [ 'Key Vault Secrest Officer' ]
    }
    {
        principalId: [ '00001-000-000-000-000000' ]
        roles: [ 'Key Vault Secrets Officer', 'Key Vault Secrets User' ]
    }
]

module rbacAssignments 'br/public:security/keyvault:0.0.1' = [ rbacAssignment in rbacAssignments: {
  name: 'rbac-assignment-guid(rbacAssignment.principalId, rbacAssignment.role)'
  params: {
    location: location
    name: keyVault.outputs.name
    newOrExisting: 'existing'
    rbacPolicies: rbacAssignment.principalIds
    roleAssignments: rbacAssignment.roles
  }
}]
```